utils/get_test_reports.py

```diff
@@ -0,0 +1,247 @@
+    """
+    relative_path = subdir.relative_to(root_test_dir)
+    report_name = f"{machine_type}_{suite}_{relative_path}_test_reports"
+    print(f"Suite: {suite} | Running on: {relative_path}")
+
+    cmd = ["python3", "-m", "pytest", "-rsfE", "-v", f"--make-reports={report_name}", str(subdir)]
+    if not cpu_tests:
+        cmd = cmd + ["-m", "not not_device_test"]
+
+    ctx_manager = tempfile.TemporaryDirectory(prefix=tmp_cache) if tmp_cache else contextlib.nullcontext()
+    with ctx_manager as tmp_dir:
+        env = os.environ.copy()
+        if tmp_cache:
+            env["HUGGINGFACE_HUB_CACHE"] = tmp_dir
+
+            print(f"Using temporary cache located at {tmp_dir = }")
+
+        print("Command:", " ".join(cmd))
+        if not dry_run:
+            subprocess.run(cmd, check=False, env=env)
+
+
+def handle_suite(
+    suite: str,
+    test_root: Path,
+    machine_type: str,
+    dry_run: bool,
+    tmp_cache: str = "",
+    resume_at: Optional[str] = None,
+    only_in: Optional[list[str]] = None,
+    cpu_tests: bool = False,
+    process_id: int = 1,
+    total_processes: int = 1,
+) -> None:
+    """
+    Handle execution of a complete test suite with advanced filtering and process distribution.
+    Args:
+        - suite (str): Name of the test suite to run (corresponds to a directory under test_root).
+        - test_root (Path): Root directory containing all test suites.
+        - machine_type (str): Machine/environment type for report naming and identification.
+        - dry_run (bool): If True, only print commands without executing them.
+        - tmp_cache (str, optional): Prefix for temporary cache directories. If empty, no temp cache is used.
+        - resume_at (str, optional): Resume execution starting from this subdirectory name.
+            Useful for restarting interrupted test runs. Defaults to None (run from the beginning).
+        - only_in (list[str], optional): Only run tests in these specific subdirectories.
+            Can include special values like IMPORTANT_MODELS. Defaults to None (run all tests).
+        - cpu_tests (bool, optional): Whether to include CPU-only tests. Defaults to False.
+        - process_id (int, optional): Current process ID for parallel execution (1-indexed). Defaults to 1.
+        - total_processes (int, optional): Total number of parallel processes. Defaults to 1.
+    """
+    # Check path to suite
+    full_path = test_root / suite
+    if not full_path.exists():
+        print(f"Test folder does not exist: {full_path}")
+        return
+
+    # Establish the list of subdir to go through
+    subdirs = sorted(full_path.iterdir())
+    subdirs = [s for s in subdirs if is_valid_test_dir(s)]
+    if resume_at is not None:
+        subdirs = [s for s in subdirs if s.name >= resume_at]
+    if only_in is not None:
+        subdirs = [s for s in subdirs if s.name in only_in]
+    if subdirs and total_processes > 1:
+        subdirs = subdirs[process_id::total_processes]
```

Comment:
I think this is not correct. You probably want to compute `n_dirs_per_process = len(subdirs) // total_processes` , then

> (process_id) * n_dirs_per_process : (process_id + 1) * n_dirs_per_process

But you also need to think of 0-based or 1-based index. If `1-based`, you will miss the first batch